File: output/qwen_2.5_72b_instruct/avatar/algo_based_translation/java/python/codeforces_30_A.py

import math

class Codeforces30A:
    def main(self):
        self.solve(input())

    def solve(self, input_string):
        import io
        sc = io.StringIO(input_string)
        A = int(sc.readline().strip())
        B = int(sc.readline().strip())
        n = int(sc.readline().strip())
        if A == 0:
            if B == 0:
                print(1)
            else:
                print("No solution")
            return
        if B % A != 0:
            print("No solution")
            return
        B //= A
        neg = B < 0
        if neg and n % 2 == 0:
            print("No solution")
            return
        if neg:
            B = -B
        for x in range(B + 1):
            if math.pow(x, n) == B:
                print(-x if neg else x)
                return
        print("No solution")

if __name__ == "__main__":
    Codeforces30A().main()

File: output/qwen_2.5_72b_instruct/avatar/direct_translation/python/java/failed_translation/codeforces_30_A.java

import java.util.Scanner;

public class codeforces_30_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int A = scanner.nextInt();
        int B = scanner.nextInt();
        int n = scanner.nextInt();
        System.out.println(task4(A, B, n));
    }

    public static String task4(int A, int B, int n) {
        for (int X = -1000; X <= 1000; X++) {
            if (A * Math.pow(X, n) == B) {
                return String.valueOf(X);
            }
        }
        return "No solution";
    }
}

